libs/remix-ui/run-tab/src/lib/actions/events.ts

```diff
@@ -45,6 +45,9 @@ export const setupEvents = (plugin: RunTab) => {
     }
     setFinalContext(plugin, dispatch)
     fillAccountsList(plugin, dispatch)
+    // 'contextChanged' is triggered on workspace change
+    // Load pinned contracts for updated workspace
+    if (context.startsWith('vm')) await loadPinnedContracts(plugin, dispatch, context)
```

Comment:
why do you check only this? we also have pinned contracts for other providers right?

Reply:
what happen if you are using an injected provider?